scim/data_groups.go

```diff
@@ -0,0 +1,34 @@
+package scim
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+)
+
+// DataSourceGroups searches from groups based on filter
+func DataSourceGroups() common.Resource {
+	type groupData struct {
+		DisplayNameFilter string   `json:"filter,omitempty"`
+		DisplayNames      []string `json:"display_names,omitempty" tf:"computed,slice_set"`
+	}
+
+	return common.WorkspaceData(func(ctx context.Context, data *groupData, w *databricks.WorkspaceClient) error {
+		groupSearch := iam.ListGroupsRequest{}
```

Comment:
You need to specify that you want to return only display names, otherwise it will fetch all attributes and will fail on big workspaces.

Also, you need to configure page size.

Reply:
Done